Add docs on migrating from Managed Service for TimescaleDB to Timescale Cloud (#756)
timescale/docs: cloud/migrate-mst-cloud/entire-database.md

1.  Connect to your Timescale Cloud database using your Timescale Cloud
    connection details. When you are prompted for a password, use your Timescale
    Cloud credentials:
    ```sql
    psql “postgres://tsdbadmin:<cloud_password>@<cloud_host>:<cloud_port>/tsdb?sslmode=require”
    ```
1.  Prepare your Timescale Cloud database for data restoration using
    [`timescaledb_pre_restore`][timescaledb_pre_restore] to stop background
    workers:
    ```sql
    SELECT timescaledb_pre_restore();
    ```
1.  At the command prompt, restore the dumped data from the `.bak` file into
    your Timescale Cloud database, using your Timescale Cloud connection
    details. To avoid permissions errors, include the `--no-owner` flag:
    ```sql
    pg_restore -U tsdbadmin -W \
    -h <cloud_host> -p <cloud_port> --no-owner \
    -Fc -v -d tsdb <file-name>.bak
    ```
1.  At the `psql` prompt, return your Timescale Cloud database to normal
    operations with the [`timescaledb_post_restore`][timescaledb_post_restore]
    command:
    ```sql
    SELECT timescaledb_post_restore();
    ```

</procedure>

## Troubleshooting
If you see any of these errors during the migration process, you can safely

> ignore them.  The migration still occurs successfully.

Reviewer: ```suggestion
ignore them. The migration still occurs successfully.
```

    Timescale Cloud. For more information, see the [list of compatible 
    extensions][extensions]. Install your other PostgreSQL extensions.
*   Checked that you're running the same major version of PostgreSQL on both
    Timescale Cloud and your source database. For information about upgrading
    PostgreSQL on your source database, see the [upgrade instructions for
    self-hosted TimescaleDB][upgrading-postgresql-self-hosted] and [Managed
    Service for TimescaleDB][upgrading-postgresql].
*   Checked that you're running the same major version of TimescaleDB on both
    Timescale Cloud and your source database. For more information, see the
    [upgrading TimescaleDB section][upgrading-timescaledb].

<highlight type="note">
To speed up migration, compress your data. You can compress any chunks where
data is not being currently inserted, updated, or deleted. When you finish the
migration, you can decompress chunks as needed for normal operation. For more
information about compression and decompression, see the
[compression section](https://docs.timescale.com/timescaledb/latest/how-to-guides/compression/).
</highlight>

<procedure>

### Migrating the entire database at once

> 1.  Dump all the data from your source database into a `.bak` file, using your

Reviewer 2: ```suggestion
1.  Dump all the data from your source database into a `dump.bak` file, using your
```

    source database connection details. If you are prompted for a password, use
    your source database credentials:
    ```bash
    pg_dump -U <SOURCE_DB_USERNAME> -W \
    -h <SOURCE_DB_HOST> -p <SOURCE_DB_PORT> -Fc -v \

>     -f <FILENAME>.bak <SOURCE_DB_NAME>

Reviewer 2: ```suggestion
    -f dump.bak <SOURCE_DB_NAME>
```

let's prefill what we can prefill to reduce number of steps for user
the filename is something we can define upfront

    ```
1.  Connect to your Timescale Cloud database using your Timescale Cloud
    connection details. When you are prompted for a password, use your Timescale
    Cloud credentials:
    ```bash
    psql “postgres://tsdbadmin:<CLOUD_PASSWORD>@<CLOUD_HOST>:<CLOUD_PORT>/tsdb?sslmode=require”
    ```
1.  Prepare your Timescale Cloud database for data restoration by using
    [`timescaledb_pre_restore`][timescaledb_pre_restore] to stop background
    workers:
    ```sql
    SELECT timescaledb_pre_restore();
    ```

> 1.  At the command prompt, restore the dumped data from the `.bak` file into

Reviewer 2: ```suggestion
1.  At the command prompt, restore the dumped data from the `dump.bak` file into
```

    your Timescale Cloud database, using your Timescale Cloud connection
    details. To avoid permissions errors, include the `--no-owner` flag:
    ```bash
    pg_restore -U tsdbadmin -W \
    -h <CLOUD_HOST> -p <CLOUD_PORT> --no-owner \

>     -Fc -v -d tsdb <FILENAME>.bak

Reviewer 2: ```suggestion
    -Fc -v -d tsdb dump.bak
```

# Migrate entire database at once

Migrate smaller databases by dumping and restoring the entire database at once.

<highlight type="warning"> 
Depending on your database size and  network speed,
migration may take several hours. You can continue reading from your old
database during this time, though performance may be slower. Writes made to your
old database during migration might not be copied to Timescale Cloud. Plan your migration accordingly to avoid data loss. 
</highlight>

## Prerequisites

- Check that the PostgreSQL utilities [`pg_dump`][pg_dump] and
  [`pg_restore`][pg_restore] are installed:

  ```bash
  pg_dump --version
  pg_restore --version

>   ```

Reviewer 2: these are two separate commands. I'm not sure if that's the way we visualize it in our docs but would suggest to break it up

  ```bash
  pg_dump --version
  ```

  ```bash
  pg_restore --version
  ```

Reviewer 2: same applies to Strategy 2

- Create a new empty database in Timescale Cloud. For instructions on installing
  Timescale Cloud, see [Install Timescale Cloud][install-timescale-cloud].

- Check that you're running the same major version of PostgreSQL on both Managed
  Service for TimescaleDB and Timescale Cloud. To upgrade your version, refer to
  the section on [upgrading PostgreSQL][upgrading-postgresql].

- Check that you're running the same major version of TimescaleDB on both
  Managed Service for TimescaleDB and Timescale Cloud. To upgrade your version,

>   refer to the section on [upgrading TimescaleDB][upgrading-timescaledb].

Reviewer 2: Upgrades for managed services are done from the Console directly.
For MST: https://kb-managed.timescale.com/en/articles/5368016-perform-a-postgresql-major-version-upgrade
For Cloud: Currently users can't upgrade/downgrade versions

Reviewer 2: same applies for Strategy 2

Author: Thanks! That link is for _PostgreSQL_ upgrades on MST, right? (It doesn't cover TimescaleDB upgrades) Is the current link for TimescaleDB upgrades correct? 

(Also, does this mean we need to update the upgrading guides in our current docs? There's no information aside from the self-hosted update instructions I originally linked.)

Reviewer 2: the docs for upgrading TimescaleDB should be fine
Cloud/MST users can do that from within the database

regarding your section question, I think someone from the docs team needs to answer this. I think there are a lot of articles in the MST knowledge base that aren't covered in our docs, and I'm not sure if we want to have them in our docs.

Author: ok I'll open a separate issue to discuss that with the docs team and mark this resolved then

<highlight type="note">
To speed up migration, compress your data. You can
compress any chunks where data is not being currently inserted, updated, or
deleted. After you finish migrating, decompress chunks as necessary for
normal operation. For more information about compression and decompression, see
the
[compression](https://docs.timescale.com/timescaledb/latest/how-to-guides/compression/)
section. 
</highlight>

## Migrating the entire database at once

<procedure>

1. Dump all the data from your old database into a .bak file:

   ```bash
   # Use your Managed Service for TimescaleDB connection details in place 
   # of <mst_host> and <mst_port>.
   pg_dump -U tsdbadmin -W -h <mst_host> -p <mst_port> -Fc -v \
   -f <file-name>.bak defaultdb
   ```

   The service prompts you for a password. Use your credentials for Managed
   Service in TimescaleDB. Wait for `pg_dump` to finish before going to the next
   step.

> 1. Connect to your Timescale Cloud database using psql:

Reviewer 2: ```suggestion
1. Connect to your Timescale Cloud database using psql or any other Postgres client:
```

they don't necessarily have to use psql

# Migrate entire database at once

Migrate smaller databases by dumping and restoring the entire database at once.

<highlight type="warning"> 
Depending on your database size and  network speed,
migration may take several hours. You can continue reading from your old
database during this time, though performance may be slower. Writes made to your
old database during migration might not be copied to Timescale Cloud. Plan your migration accordingly to avoid data loss. 
</highlight>

## Prerequisites

- Check that the PostgreSQL utilities [`pg_dump`][pg_dump] and
  [`pg_restore`][pg_restore] are installed:

  ```bash
  pg_dump --version
  ```

  ```bash
  pg_restore --version
  ```

- Check that you have a client for connecting to PostgreSQL. These instructions

>   assume you are using [psql][psql], but any client will work.

Author: ```suggestion
  assume that you're using [psql][psql], but any client will work.
```

- Create a new empty database in Timescale Cloud. For instructions on installing
  Timescale Cloud, see [Install Timescale Cloud][install-timescale-cloud].

> <highlight type="tip">For faster migration, [compress all your

Reviewer: Use note, important, or warning. Tip is only available for backwards compatibility.

> <highlight type="tip">For faster migration, [compress all your
> data](https://docs.timescale.com/timescaledb/latest/how-to-guides/compression/manually-compress-chunks/)

Reviewer: ```suggestion
<highlight type="tip">
To speed up migration, compress your data. For more information about 
compression, see the 
[compression section](https://docs.timescale.com/timescaledb/latest/how-to-guides/compression/manually-compress-chunks/)
</highlight>
```

    SELECT timescaledb_post_restore();
    ```

</procedure>

Your data is now in Timescale Cloud. If you compressed it before migration,

> [decompress chunks][decompress] as necessary for normal operation.

Reviewer: I like to try and avoid result statements. Can we reframe this in the "for more information, see" format instead?
